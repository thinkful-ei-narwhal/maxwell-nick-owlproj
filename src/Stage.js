import React from 'react';
import OnStageParticipant from './OnStageParticipant';

import './stage.css'

export default function Stage(props) {
  const onStageParticipants = [];
  props.participants.forEach(participant => {
    if (participant.onStage) {
      onStageParticipants.push(<OnStageParticipant key={participant.id} name={participant.name} avatar={participant.avatar} />);
    }
  });

  return (
    <div className='stage'>
      {onStageParticipants}
    </div>
  )
}
