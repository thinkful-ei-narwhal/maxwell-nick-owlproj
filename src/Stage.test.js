import React from 'react';
import ReactDOM from 'react-dom';
import errorRenderer from 'react-test-renderer';
import store from './store';
import Stage from './Stage';

/* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST */
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Stage participants={store.participants} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

/* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST */

it('renders the UI as expected', () => {
  const JSON = errorRenderer
    .create(<Stage participants={store.participants} />)
    .toJSON();
  expect(JSON).toMatchSnapshot();
});