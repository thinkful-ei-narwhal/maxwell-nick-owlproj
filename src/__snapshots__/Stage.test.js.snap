// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="stage"
>
  <div
    className="onStage-participant"
  >
    <header>
      <h1>
        Koren Templeton
      </h1>
    </header>
    <img
      alt="testing"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="onStage-participant"
  >
    <header>
      <h1>
        Ashla Attwool
      </h1>
    </header>
    <img
      alt="testing"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
